src/txn/txn_timestamp.c

```diff
@@ -187,6 +187,12 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
 
         WT_STAT_CONN_INCR(session, txn_walk_sessions);
         WT_STAT_CONN_INCRV(session, txn_sessions_walked, i);
+    } else if (WT_CONFIG_LIT_MATCH("backup_checkpoint", cval)) {
+        /* Read-only value forever. Make sure we don't used a cached version. */
+        ts = WT_TS_NONE;
```

Comment:
This implementation still requires the atomic, can we not extend the scope of the lock to include the if statement? Mixing reads of variables written to within locks is not a sound pattern.

Reply:
Actually the `WT_WITH_HOTBACKUP_READ_LOCK` doesn't work (test fails) because that macro only performs the operation when backup is not also in progress. So I need to rename/repurpose that macro a bit.

Reply:
It may be worth adding a comment that says something like:
This will only set the timestamp if the backup is running.